src/Diagnostics.RuntimeHost/Controllers/ApiManagement/ApiManagementController.cs

```diff
@@ -35,10 +35,16 @@ public ApiManagementController(IStampService stampService, ICompilerHostClient c
             return await base.GetDetector(GetResource(subscriptionId, resourceGroupName, serviceName), detectorId, startTime, endTime, timeGrain);
         }
 
+        [HttpPost(UriElements.Detectors + UriElements.DetectorResource + UriElements.StatisticsQuery)]
+        public async Task<IActionResult> ExecuteSystemQuery(string subscriptionId, string resourceGroupName, string serviceName, [FromBody]CompilationBostBody<dynamic> jsonBody, string detectorId, string startTime = null, string endTime = null, string timeGrain = null)
+        {
+            return await base.ExecuteQuery(GetResource(subscriptionId, resourceGroupName, serviceName), jsonBody, startTime, endTime, timeGrain, detectorId);
+        }
+
         [HttpPost(UriElements.Detectors + UriElements.DetectorResource + UriElements.Statistics + UriElements.StatisticsResource)]
-        public async Task<IActionResult> GetSystemInvoker(string subscriptionId, string resourceGroupName, string serviceName,  string detectorId, string invokerId, string startTime = null, string endTime = null, string timeGrain = null)
+        public async Task<IActionResult> GetSystemInvoker(string detectorId, string invokerId)
```

Comment:
You would need to add back these parameters as they are part of the Common Route at the class level

Reply:
Oh.. I see.. So all the parameters in the exact order with routing path right?

Reply:
correct